### ISSUE TITLE
Adaptation to Active Learning

### ISSUE THREAD

**Issue by nhaH-luaP** [microsoft/Semi-supervised-learning#56]
I have implemented some of the methods you presented (pseudo-labels, pi-model, fixmatch) with the goal in mind of using them in an active learning setting. I am currently using a non-pretrained WRS2810. However, the current biggest obstacle is the training time. E.g. the training of fixmatch for 1024 epochs á 1024 train steps (1048576 steps total) with an uratio of 7 (example from CIFAR10 with 40 labeled samples) takes a very long time and would additionally have to be repeated several times in an active learning environment, which makes the whole thing unusable. My questions are therefore:

- Is ViT and pretaining a better choice for this purpose?
- Why is it necessary to train so many epochs?
- Is there an alternative training method that reduces training time while causing minimal performance degradation?
- What do you think about applying these methods to active learning? Is your approach to training adaptable to this scenario?

Thanks for your help!
Paul

**Comment by Hhhhhhao** [comment]
Hi, it is definitely possible to adapt current algorithms to active learning and I believe there are active research topics about this. 

Regard your questions:
1. Yes. Pretrained ViT can reduce requirement of the number of training iterations and less unlabeled ratio.
2. It is set by Google's FixMatch (and earlier papers) as a default I think. 
3. Currently there is no other accelerating methods in my mind. Maybe you can try mixed precision.
4. Absolutely adaptable to active learning, but I'm not an expert in that area and I don't know how yet.
